src/ua_securechannel.c

```diff
@@ -1299,7 +1299,14 @@ checkSymHeader(UA_SecureChannel *const channel,
             else
                 return UA_STATUSCODE_BADSECURECHANNELTOKENUNKNOWN;
         }
-        return UA_SecureChannel_revolveTokens(channel);
+        UA_StatusCode retval = UA_SecureChannel_revolveTokens(channel);
+        if(retval != UA_STATUSCODE_GOOD)
+            return retval;
+        if(channel->securityToken.tokenId == tokenId) {
```

Comment:
The next guy will scratch his head looking at this.

Can you add a few comments what is happening here?
Under which conditions may `channel->securityToken.tokenId == tokenId`?